File: CollaborationBack/src/main/java/com/collab/CollaborationBack/Service/UserService.java
package com.collab.CollaborationBack.Service;


import java.util.List;

import com.collab.CollaborationBack.model.User;

public interface UserService {
	
boolean registeruser(User user);
boolean isvalidemail(String emailId);
boolean isvalidusername(String username);
User login(User user);
void update(User user);
User getuser(String username);
List<User> getallusers();	
	
}

File: CollaborationFront/User/userservice.js
/**
 * user service
 */

app.factory('UserService',function($http){

	var userService={}
	var BASE_URL="http://localhost/CollaborationMiddle"
		userService.registerUser=function(user)
		{
		  return $http.post(BASE_URL + "/registeruser",user)
	    }
	    userService.login=function(user)
	    {
		  return $http.post(BASE_URL + "/login",user)
	    }
	
	    userService.logout=function(){
	    	return $http.put(BASE_URL+"/logout")
	    }
	    userService.getUser=function(){
	    	return $http.get(BASE_URL+"/getuser") 
	    }
	    
	    
	    return userService;
	
})